consensus/state.go

```diff
@@ -1295,6 +1295,37 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 		return
 	}
 
+	// Determine if the proposed block has a sane, non-nil proof-of-lock.
```

Comment:
Might be helpful to cite the original original here in comment (and mention we skip `valid(v)`):
```
		28: upon <PROPOSAL, h_p, round_p, v, v_r> from proposer(h_p, round_p) AND 2f + 1 <PREVOTE, h_p, v_r, id(v)> while
		step_p = propose && (v_r ≥ 0 && v_r < round_p) do
		29: if valid(v) && (lockedRound_p ≤ v_r || lockedValue_p = v) then
		30: broadcast <PREVOTE, h_p, round_p, id(v)>
```

Reply:
Added comments